data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -253,8 +255,8 @@ private static Schema fileProjection(Schema tableSchema, Schema requestedSchema,
     // TODO: support adding nested columns. this will currently fail when finding nested columns to add
     List<Types.NestedField> columns = Lists.newArrayList(requestedSchema.columns());
     for (int fieldId : missingIds) {
```

Comment:
should we add this new metadata column to `requiredIds` above?

Reply:
I think the logic here is if the `requestedSchema` doesn't contain the metadata column, we add them to the end. the `requiredIds` already has metadata columns.